templates/Vagrantfile.tox

```diff
@@ -0,0 +1,40 @@
+# -*- mode: ruby -*-
+# vi: set ft=ruby :
+#
+
+Vagrant.configure(2) do |config|
+
+    config.ssh.username = "root"
+
+    config.vm.synced_folder "./", "/vagrant",
+        type: "nfs",
+        nfs_udp: false
+
+    config.vm.box = "{{ vagrant_template_name }}"
+    config.vm.box_version = "{{ vagrant_template_version }}"
+
+    config.vm.provider "libvirt" do |domain, override|
+        # Nested virtualization options
+        domain.nested = true
+        domain.cpu_mode = "host-passthrough"
+
+        # Disable graphics
+        domain.graphics_type = "none"
+
+        # Recommended for remote NFS storage
+        domain.volume_cache = "none"
+    end
+
+    config.vm.define "builder"  do |builder|
+        builder.vm.provider "libvirt" do |domain,override|
+            domain.cpus = 4
```

Comment:
I'm sure tox testing isn't that resource intenstive. How about 1 CPU / 900 MiB RAM?